src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -213,6 +216,16 @@ class WorkspaceService(dataSource: DataSource, workspaceDAO: WorkspaceDAO, entit
       }
     }
 
+  def evaluateExpression(workspaceNamespace: String, workspaceName: String, entityType: String, entityName: String, expression: String): PerRequestMessage =
+    dataSource inTransaction { txn =>
+      txn withGraph { graph =>
+        new ExpressionEvaluator( graph, new ExpressionParser() ).evaluate(workspaceNamespace, workspaceName, entityType, entityName, expression) match {
+          case Success(result) => RequestComplete(http.StatusCodes.OK, result.map(AttributeConversions.propertyToAttribute))
+          case Failure(regret) => RequestComplete(http.StatusCodes.BadRequest, regret.getMessage)
```

Comment:
do we want to distinguish between internal errors and bad request? Are there any of the former?